src/transaction/transaction.ts

```diff
@@ -112,67 +126,21 @@ class Transaction {
     boundParams: BoundParams;
   }> {
     const merkleTree = wallet.merkletree[this.chainID];
-    const merkleRoot = await merkleTree.getRoot(0);
+    const merkleRoot = await merkleTree.getRoot(this.spendingTree); // TODO: Is this correct tree?
```

Comment:
@mesquka 
Is this the correct tree? Spending tree?
It was hardcoded as 0 before.

Reply:
Yes, hardcode to 0 will only spend from the first merkle tree.